build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -414,4 +426,41 @@ public boolean visit(IResourceProxy proxy) throws CoreException {
 
 	}
 
+	private String findCompilerInPath(ITool tool, IConfiguration config) {
+		if (toolMap.containsKey(tool)) {
```

Comment:
Can `CCorePlugin.getDefault().getBuildEnvironmentManager().getVariables(cfg, true)` can return different values over the lifetime of the toolMap cache?

Reply:
No, it can't. That's why the check is done before we look up the environment. I've addded a comment on the variable.

Reply:
Great, thanks for confirming and documenting.